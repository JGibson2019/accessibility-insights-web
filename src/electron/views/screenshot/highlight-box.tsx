// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { BoundingRectangle } from 'electron/platform/android/scan-results';
import { highlightBox, highlightBoxLabel } from 'electron/views/screenshot/highlight-box.scss';
import * as React from 'react';
import { CSSProperties } from 'react';

export interface HighlightBoxProps {
    boundingRectangle: BoundingRectangle;
}

export const HighlightBox = NamedFC<HighlightBoxProps>('HighlightBox', props => {
    const { boundingRectangle } = props;

    const width = boundingRectangle.right - boundingRectangle.left;
    const height = boundingRectangle.bottom - boundingRectangle.top;

    const boxStyles: CSSProperties = {
        width: `${1.1 * width}px`,
        height: `${1.1 * height}px`,
        top: `${0.98 * boundingRectangle.top}px`,
        left: `${0.9 * boundingRectangle.left}px`,
    };

    return (
        <div className={highlightBox} style={boxStyles} ariaHidden="true">
            <div className={highlightBoxLabel}>!</div>
        </div>
    );
});
