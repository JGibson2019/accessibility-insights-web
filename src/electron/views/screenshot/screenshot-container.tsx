// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { ScreenshotData } from 'common/types/store-data/unified-data-interface';
import { BoundingRectangle } from 'electron/platform/android/scan-results';
import { HighlightBox } from 'electron/views/screenshot/highlight-box';
import { Screenshot } from 'electron/views/screenshot/screenshot';
import { screenshotContainer } from 'electron/views/screenshot/screenshot-container.scss';
import * as React from 'react';

export interface ScreenshotContainerDeps {}
export interface ScreenshotContainerProps {
    deps?: ScreenshotContainerDeps;
    screenshotData: ScreenshotData;
}

const boundingRectangle: BoundingRectangle = {
    top: 0,
    bottom: 100,
    left: 0,
    right: 100,
};

export const ScreenshotContainer = NamedFC<ScreenshotContainerProps>('ScreenshotContainer', props => {
    return (
        <div className={screenshotContainer}>
            <Screenshot encodedImage={props.screenshotData.base64PngData} altText={'test'} />
            <HighlightBox deps={null} boundingRectangle={boundingRectangle} />
        </div>
    );
});
