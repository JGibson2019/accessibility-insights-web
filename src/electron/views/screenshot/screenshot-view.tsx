// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { ScreenshotData } from 'common/types/store-data/unified-data-interface';
import { isEmpty, isUndefined } from 'lodash';
import * as React from 'react';

import { Screenshot } from './screenshot';

export type ScreenshotViewProps = {
    screenshotData: ScreenshotData;
};

const screenshotAltText: string = 'axe-android results screenshot with highlighted components';

export const ScreenshotView = NamedFC<ScreenshotViewProps>('ScreenshotView', (props: ScreenshotViewProps) => {
    if (isUndefined(props.screenshotData) || isEmpty(props.screenshotData)) {
        return (
            <div>
                <p>Screenshot for scan is unavailable</p>
            </div>
        );
    } else {
        return (
            <div>
                <h1 tabIndex={0}>Page state screenshot</h1>
                <Screenshot encodedImage={props.screenshotData.base64PngData} altText={screenshotAltText} />
            </div>
        );
    }
});
