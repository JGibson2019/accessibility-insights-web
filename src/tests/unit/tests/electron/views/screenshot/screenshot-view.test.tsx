// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ScreenshotView, ScreenshotViewProps } from 'electron/views/screenshot/screenshot-view';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('ScreenshotView', () => {
    describe('renders', () => {
        it('when passed a value for screenshotData', () => {
            const props: ScreenshotViewProps = {
                screenshotData: { base64PngData: 'string' },
            };
            const wrapper = shallow(<ScreenshotView {...props} />);

            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('when passed an empty value for screenshotData', () => {
            const props = {} as ScreenshotViewProps;

            const wrapper = shallow(<ScreenshotView {...props} />);

            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
