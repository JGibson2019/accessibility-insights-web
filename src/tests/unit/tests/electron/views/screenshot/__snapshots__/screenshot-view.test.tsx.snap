// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenshotView renders when passed a value for screenshotData 1`] = `
<div>
  <h1
    tabIndex={0}
  >
    Target Page Screenshot
  </h1>
  <ScreenshotContainer
    screenshotData={
      Object {
        "base64PngData": "test-base-64-png-data",
      }
    }
  />
</div>
`;

exports[`ScreenshotView renders when passed an empty value for screenshotData 1`] = `
<div>
  <h1
    tabIndex={0}
  >
    Target Page Screenshot
  </h1>
  <ScreenshotContainer
    screenshotData={Object {}}
  />
</div>
`;

exports[`ScreenshotView renders when passed an null value for screenshotData 1`] = `
<div>
  <h1
    tabIndex={0}
  >
    Target Page Screenshot
  </h1>
  <ScreenshotContainer
    screenshotData={null}
  />
</div>
`;

exports[`ScreenshotView renders when passed an undefined value for screenshotData 1`] = `
<div>
  <h1
    tabIndex={0}
  >
    Target Page Screenshot
  </h1>
  <ScreenshotContainer />
</div>
`;
